mayastor/src/nexus_uri.rs

```diff
@@ -83,6 +83,21 @@ pub fn bdev_get_name(uri: &str) -> Result<String, NexusBdevError> {
     Ok(Uri::parse(uri)?.get_name())
 }
 
+impl std::cmp::PartialEq<url::Url> for &Bdev {
+    fn eq(&self, uri: &url::Url) -> bool {
+        match Uri::parse(&uri.to_string()) {
+            Ok(device) if device.get_name() == self.name() => {
+                self.driver()
+                    == match uri.scheme() {
+                        "nvmf" | "pcie" => "nvme",
+                        scheme => scheme,
+                    }
+            }
+            _ => false,
+        }
+    }
+}
+
 impl std::cmp::PartialEq<url::Url> for Bdev {
     fn eq(&self, uri: &url::Url) -> bool {
         match Uri::parse(&uri.to_string()) {
```

Comment:
since you've added the trait above, this can just become self == uri ?

Reply:
I'm not sure what you are suggesting..

Reply:
You've added the trait PartialEq for &Bdev. So now the trait PartialEq for Bdev could use make use of the &Bdev version becoming a simple bdev == uri on line 103?

Reply:
as discussed we cant :+1: